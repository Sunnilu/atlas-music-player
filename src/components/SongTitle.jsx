// src/components/SongTitle.jsx
export default function SongTitle() {
  return (
    <div className="text-center mb-4">
      <h2 className="text-2xl font-bold text-gray-900 mb-2">Song Title</h2>
      <p className="text-gray-600">Artist Name</p>
    </div>
  );
}