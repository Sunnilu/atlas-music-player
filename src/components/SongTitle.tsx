const SongTitle = ({ song }) => {
  if (!song) return <p className="text-white">No song selected</p>;
  return (
    <div className="text-center text-white">
      <p className="font-bold">{song.title}</p>
      <p className="text-sm text-gray-300">{song.artist}</p>
    </div>
  );
};
