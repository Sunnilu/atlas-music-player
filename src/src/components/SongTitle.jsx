import React from 'react';

const SongTitle = ({ title = "Song Title", author = "Artist Name" }) => {
  return (
    <div className="flex flex-col">
      <span className="text-sm font-semibold text-white truncate">{title}</span>
      <span className="text-xs text-gray-400 truncate">{author}</span>
    </div>
  );
};

export default SongTitle;
