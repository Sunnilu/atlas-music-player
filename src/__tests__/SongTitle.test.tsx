// src/__tests__/SongTitle.test.tsx
import { describe, it, expect } from 'vitest';
import renderer from 'react-test-renderer';
import SongTitle from '../components/SongTitle';

describe('SongTitle Snapshot', () => {
  it('renders with standard title and artist', () => {
    const tree = renderer
      .create(<SongTitle title="Ocean Eyes" artist="Billie Eilish" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with long title and artist', () => {
    const tree = renderer
      .create(
        <SongTitle
          title="A Very Long Song Title That Might Break Lines"
          artist="An Even Longer Artist Name That Might Wrap"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with empty title and artist', () => {
    const tree = renderer
      .create(<SongTitle title="" artist="" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
