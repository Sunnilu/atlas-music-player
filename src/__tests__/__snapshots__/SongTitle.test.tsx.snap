// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Snapshot > renders with empty title and artist 1`] = `null`;

exports[`SongTitle Snapshot > renders with long title and artist 1`] = `null`;

exports[`SongTitle Snapshot > renders with standard title and artist 1`] = `null`;
